modules/solid_properties/include/materials/ThermalFunctionSolidProperties.h

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ThermalSolidPropertiesMaterial.h"
+#include "Function.h"
```

Comment:
Just forward declaring this should work

Reply:
It didn't seem to work for me

Reply:
What's the error?! All you have are Function references? 

```
class Function;
```

should be more than enough with a proper include in the source file

Reply:
Oh, I forgot to do an include in the source file